catalog/provider.go

```diff
@@ -37,31 +42,37 @@ var _ configuration.DataDirProvider = (*DatabaseProvider)(nil)
 const readOnlySuffix = "?access_mode=read_only"
 
 func NewInMemoryDBProvider() *DatabaseProvider {
-	prov, err := NewDBProvider(".", "")
+	prov, err := NewDBProvider("", ".", "")
 	if err != nil {
 		panic(err)
 	}
 	return prov
 }
 
-func NewDBProvider(dataDir, dbFile string) (*DatabaseProvider, error) {
-	dbFile = strings.TrimSpace(dbFile)
-	name := ""
-	dsn := ""
-	if dbFile == "" {
-		// in-memory mode, mainly for testing
-		name = "memory"
+func NewDBProvider(defaultTimeZone, dataDir, defaultDB string) (prov *DatabaseProvider, err error) {
```

Comment:
It seems we don't have strategy to scan all .db files under the data dir and attach them to current datbase?